File: Data Structures & Algorithms/is-palindrome/submission-2.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string t = "";
        for(int i = 0; i < s.size(); i++){
            if('0' <= s[i] && s[i] <= '9'){
                t += s[i];
            }
            if('a' <= s[i] && s[i] <= 'z'){
                t += s[i];
            }
            else if('A' <= s[i]  && s[i] <= 'Z') {
                t += (s[i] - 'A' + 'a');
            }
        }
       
        for(int i = 0; i < t.size()/2; i++){
            if(t[i] != t[t.size() -i -1]) {
                return false;
            }
        }
        return true;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder sb = new StringBuilder();
        for (String s : strs) {
            sb.append(s.length());
            sb.append(" ");
            sb.append(s);
        }

        return sb.toString();
    }

    public List<String> decode(String str) {
        List<String> strs = new ArrayList<>();

        int i = 0;
        while (i < str.length()) {
            int j = i;
            while (str.charAt(j) != ' ') {
                j++;
            }

            Integer len = Integer.parseInt(str.substring(i, j));

            strs.add(str.substring(j+1, len + j + 1));
            i = len + j + 1;
        }

        return strs;
    }
}
